pkg/controllers/monitoring/monitoring-stack/conditions.go

```diff
@@ -9,36 +9,45 @@ import (
 )
 
 const (
-	AvailableReason                = "MonitoringStackAvailable"
-	ReconciledReason               = "MonitoringStackReconciled"
-	FailedToReconcileReason        = "FailedToReconcile"
-	PrometheusNotAvailable         = "PrometheusNotAvailable"
-	PrometheusNotReconciled        = "PrometheusNotReconciled"
-	PrometheusDegraded             = "PrometheusDegraded"
-	ResourceSelectorIsNil          = "ResourceSelectorNil"
-	CannotReadPrometheusConditions = "Cannot read Prometheus status conditions"
-	AvailableMessage               = "Monitoring Stack is available"
-	SuccessfullyReconciledMessage  = "Monitoring Stack is successfully reconciled"
-	ResourceSelectorIsNilMessage   = "No resources will be discovered, ResourceSelector is nil"
-	ResourceDiscoveryOnMessage     = "Resource discovery is operational"
-	NoReason                       = "None"
+	AlertmanagerNotAvailable         = "AlertmanagerNotAvailable"
+	AlertmanagerDegraded             = "AlertmanagerDegraded"
+	AlertmanagerNotReconciled        = "AlertmanagerNotReconciled"
+	AvailableReason                  = "MonitoringStackAvailable"
+	ReconciledReason                 = "MonitoringStackReconciled"
+	FailedToReconcileReason          = "FailedToReconcile"
+	PrometheusNotAvailable           = "PrometheusNotAvailable"
+	PrometheusNotReconciled          = "PrometheusNotReconciled"
+	PrometheusDegraded               = "PrometheusDegraded"
+	ResourceSelectorIsNil            = "ResourceSelectorNil"
+	CannotReadPrometheusConditions   = "Cannot read Prometheus status conditions"
+	CannotReadAlertmanagerConditions = "Cannot read Alertmanager status conditions"
+	AvailableMessage                 = "Monitoring Stack is available"
+	SuccessfullyReconciledMessage    = "Monitoring Stack is successfully reconciled"
+	ResourceSelectorIsNilMessage     = "No resources will be discovered, ResourceSelector is nil"
+	ResourceDiscoveryOnMessage       = "Resource discovery is operational"
+	NoReason                         = "None"
 )
 
-func updateConditions(ms *v1alpha1.MonitoringStack, prom monv1.Prometheus, recError error) []v1alpha1.Condition {
+func updateConditions(ms *v1alpha1.MonitoringStack, prom monv1.Prometheus, alertManager *monv1.Alertmanager, recError error) []v1alpha1.Condition {
```

Comment:
nit: I think that here both prom and alertmanager should be pointers. Eventually, ObO will support other monitoring O11y signals so there might a future where Prometheus might also be disabled. I'm also happy with them both becoming references as long as we stay consistent.

Reply:
This was my initial comment as well which I deleted because this usage has a  semantic meaning. i.e. Prometheus isn't an optional component.

Reply:
I see, I had seen your comment but misunderstood the intent behind it 🤦‍♂️ and now that I think about it I'm wrong because if users provide a MonitoringStack they should at least want Prometheus right 🤔